sdk/servicebus/service-bus/review/service-bus.api.md

```diff
@@ -3,479 +3,622 @@
-    body: any;
-    contentType?: string;
-    correlationId?: string | number | Buffer;
-    label?: string;
-    messageId?: string | number | Buffer;
-    partitionKey?: string;
-    replyTo?: string;
-    replyToSessionId?: string;
-    scheduledEnqueueTimeUtc?: Date;
-    sessionId?: string;
-    timeToLive?: number;
-    to?: string;
-    userProperties?: {
-        [key: string]: any;
-    };
-    viaPartitionKey?: string;
+  body: any;
+  contentType?: string;
+  correlationId?: string | number | Buffer;
+  label?: string;
+  messageId?: string | number | Buffer;
+  partitionKey?: string;
+  replyTo?: string;
+  replyToSessionId?: string;
+  scheduledEnqueueTimeUtc?: Date;
+  sessionId?: string;
+  timeToLive?: number;
+  to?: string;
+  userProperties?: {
+    [key: string]: any;
+  };
+  viaPartitionKey?: string;
 }
 
 // @public
 export class Sender {
-    cancelScheduledMessage(sequenceNumber: Long): Promise<void>;
-    cancelScheduledMessages(sequenceNumbers: Long[]): Promise<void>;
-    close(): Promise<void>;
-    get isClosed(): boolean;
-    scheduleMessage(scheduledEnqueueTimeUtc: Date, message: SendableMessageInfo): Promise<Long>;
-    scheduleMessages(scheduledEnqueueTimeUtc: Date, messages: SendableMessageInfo[]): Promise<Long[]>;
-    send(message: SendableMessageInfo): Promise<void>;
-    sendBatch(messages: SendableMessageInfo[]): Promise<void>;
-    }
+  cancelScheduledMessage(sequenceNumber: Long, options?: OperationOptions): Promise<void>;
+  cancelScheduledMessages(sequenceNumbers: Long[], options?: OperationOptions): Promise<void>;
+  close(): Promise<void>;
+  get isClosed(): boolean;
+  scheduleMessage(
+    scheduledEnqueueTimeUtc: Date,
+    message: SendableMessageInfo,
+    options?: OperationOptions
+  ): Promise<Long>;
+  scheduleMessages(
+    scheduledEnqueueTimeUtc: Date,
+    messages: SendableMessageInfo[],
+    options?: OperationOptions
+  ): Promise<Long[]>;
+  send(message: SendableMessageInfo, options?: OperationOptions): Promise<void>;
+  sendBatch(messages: SendableMessageInfo[], options?: OperationOptions): Promise<void>;
+}
 
-// @public
+// @public (undocumented)
 export class ServiceBusClient {
-    constructor(connectionString: string, options?: ServiceBusClientOptions);
-    constructor(host: string, credential: TokenCredential, options?: ServiceBusClientOptions);
-    close(): Promise<any>;
-    createQueueClient(queueName: string): QueueClient;
-    createSubscriptionClient(topicName: string, subscriptionName: string): SubscriptionClient;
-    createTopicClient(topicName: string): TopicClient;
-    readonly name: string;
+  constructor(connectionString: string, options?: ServiceBusClientOptions);
+  constructor(
+    hostName: string,
+    tokenCredential: TokenCredential_2,
+    options?: ServiceBusClientOptions
+  );
+  // (undocumented)
+  close(): Promise<void>;
+  createReceiver(queueName: string, receiveMode: "peekLock"): NonSessionReceiver<"peekLock">;
```

Comment:
Do the order of these arguments make sense? Should 'receiveMode' go first?

Reply:
Yes. They do. "Where do you want to receive from? What mode?" My line of thinking isn't usually the opposite.

Reply:
@conniey  - Just checking - you _like_ this order or you don't like this ordering? :)

Reply:
As an aside, in .NET the receiveMode is specified on the ServiceBusReceiverOptions, along with the PrefetchCount and RetryOptions.

Reply:
This is probably a good point worth discussing. LockMode has a pretty big bearing on how you deal with messages "properly" so it might be good to make it obvious and prominent.

Perhaps it doesn't matter as much in the Processor where you're doing a lot of that on their behalf.

Reply:
The main reason for putting in the options is that there is a sensible default of PeekLock for this property, so it didn't seem worth elevating to be a parameter in the ctors.

Reply:
@JoshLove-msft  - let me throw this out there. :)

So peekLock is a good default (if you're choosing a lock mode) but it's paired with some behavior the user needs to understand if they choose any "get events" method aside from 'subscribe' (which is our callback based version).  Ie, the user has to know to complete messages, etc...

I'm wondering if we are making a bad call here by defaulting to something and not making it explicit.

Reply:
In .NET, we need to have ctor overloads absent optional parameters in addition to the ones that have the optional parameters. Now that the receiver cannot be constructed from a ctor, this is blurred a bit. As a general rule though, I still think it makes sense to have optional parameters be specified in the options bag. As you say, the documentation should make it clear what the default value is for the ReceiveMode and what its effect is on behavior.